cobigen/cobigen-core-parent/cobigen-core/src/main/java/com/devonfw/cobigen/impl/util/TemplatesClassloaderUtil.java

```diff
@@ -0,0 +1,351 @@
+     * If the sources are not compiled, the templates will not be able to be generated.
+     * @param configurationFolder
+     *            Path to add to ClassLoader
+     * @param classLoader
+     *            ClassLoader to load jar from
+     * @return a List of Classes for template generation.
+     * @throws IOException
+     *             if either templates jar or templates folder could not be read
+     */
+    public static List<Class<?>> resolveUtilClasses(Path configurationFolder, ClassLoader classLoader)
+        throws IOException {
+        List<Class<?>> result = new LinkedList<>();
+        ArrayList<URL> classLoaderUrls = new ArrayList<>(); // stores ClassLoader URLs
+        Path templateRoot = null;
+        ClassLoader inputClassLoader = null;
+        URL contextConfigurationLocation = null;
+        if (configurationFolder != null) {
+            templateRoot = configurationFolder;
+            classLoaderUrls = addFoldersToClassLoaderUrls(configurationFolder);
+            inputClassLoader = getUrlClassLoader(classLoaderUrls.toArray(new URL[] {}));
+            contextConfigurationLocation = configurationFolder.toUri().toURL();
+        } else {
+            inputClassLoader = classLoader;
+            contextConfigurationLocation = getContextConfiguration(inputClassLoader);
+        }
+
+        if (contextConfigurationLocation.toString().startsWith("jar")) {
+            result = resolveFromJar(result, inputClassLoader, contextConfigurationLocation);
+        } else {
+            result = resolveFromFolder(result, templateRoot, inputClassLoader);
+        }
+
+        return result;
+    }
+
+    /**
+     * Resolves utility classes from Folder
+     *
+     * @param result
+     *            List to store utility classes in
+     * @param templateRoot
+     *            Path to template folder containing classes
+     * @param inputClassLoader
+     *            ClassLoader to use for storing of classes
+     * @return List of classes to load utilities from
+     */
+    private static List<Class<?>> resolveFromFolder(List<Class<?>> result, Path templateRoot,
+        ClassLoader inputClassLoader) {
+        LOG.debug("Processing configuration folder " + templateRoot.toString());
+        LOG.info("Searching for classes in configuration folder...");
+        List<Path> foundPaths = new LinkedList<>();
+
+        try {
+            foundPaths = walkTemplateFolder(templateRoot);
+        } catch (IOException e) {
+            LOG.error("Could not read templates folder", e);
+        }
+        if (foundPaths.size() > 0) {
+
+            // clean up test classes
+            Iterator<Path> it = foundPaths.iterator();
+            while (it.hasNext()) {
+                Path next = it.next();
+                if (!templateRoot.relativize(next).startsWith("target/classes")) {
+                    LOG.info("    * Removed test class file {}", next);
```

Comment:
Better on debug as there might be a lot test classes

Reply:
adjusted.